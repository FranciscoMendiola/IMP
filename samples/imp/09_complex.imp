n := 5;
fact := 1;
while not (n = 0) do (
  fact := fact * n;
  n := n - 1
)
