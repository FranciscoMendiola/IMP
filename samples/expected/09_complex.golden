Token {tokenKind = Id, lexeme = "n", position = 0}
Token {tokenKind = Assign, lexeme = ":=", position = 2}
Token {tokenKind = IntLit, lexeme = "5", position = 5}
Token {tokenKind = Semi, lexeme = ";", position = 6}
Token {tokenKind = Id, lexeme = "fact", position = 8}
Token {tokenKind = Assign, lexeme = ":=", position = 13}
Token {tokenKind = IntLit, lexeme = "1", position = 16}
Token {tokenKind = Semi, lexeme = ";", position = 17}
Token {tokenKind = KWhile, lexeme = "while", position = 19}
Token {tokenKind = KNot, lexeme = "not", position = 25}
Token {tokenKind = LPar, lexeme = "(", position = 29}
Token {tokenKind = Id, lexeme = "n", position = 30}
Token {tokenKind = Eq, lexeme = "=", position = 32}
Token {tokenKind = IntLit, lexeme = "0", position = 34}
Token {tokenKind = RPar, lexeme = ")", position = 35}
Token {tokenKind = KDo, lexeme = "do", position = 37}
Token {tokenKind = LPar, lexeme = "(", position = 40}
Token {tokenKind = Id, lexeme = "fact", position = 44}
Token {tokenKind = Assign, lexeme = ":=", position = 49}
Token {tokenKind = Id, lexeme = "fact", position = 52}
Token {tokenKind = Times, lexeme = "*", position = 57}
Token {tokenKind = Id, lexeme = "n", position = 59}
Token {tokenKind = Semi, lexeme = ";", position = 60}
Token {tokenKind = Id, lexeme = "n", position = 64}
Token {tokenKind = Assign, lexeme = ":=", position = 66}
Token {tokenKind = Id, lexeme = "n", position = 69}
Token {tokenKind = Minus, lexeme = "-", position = 71}
Token {tokenKind = IntLit, lexeme = "1", position = 73}
Token {tokenKind = RPar, lexeme = ")", position = 75}
